app/src/main/java/com/arctouch/codechallenge/api/MoviesRepository.kt

```diff
@@ -0,0 +1,34 @@
+package com.arctouch.codechallenge.api
+
+import android.annotation.SuppressLint
+import com.arctouch.codechallenge.model.GenreResponse
+import com.arctouch.codechallenge.model.UpcomingMoviesResponse
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+
+//repository to centralize data request
+class MoviesRepository (private val api: TmdbApi): MoviesDataSource {
+    @SuppressLint("CheckResult")
+    override fun getMovieList(page: Long, success: (UpcomingMoviesResponse) -> Unit, failure: (Throwable) -> Unit) {
+        api.upcomingMovies(TmdbApi.API_KEY, TmdbApi.DEFAULT_LANGUAGE, page, TmdbApi.DEFAULT_REGION)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe ({
```

Comment:
Should you have kept the value returned by `subscribe()`?

Reply:
kept as cache? I don't understood

Reply:
The subscribe method returns a `Disposable` object. Is there a scenario in which you would want to use this object for anything?

Reply:
I think in this project i don't need or don't imagine a need to keep this object after the use, if that's your question, sorry if i don't understand.